src/ConfigFileParser.cpp

```diff
@@ -387,111 +613,51 @@ std::string ConfigFileParser::getValue(const std::string& line) const
- * For the case the line does not contain any more directives, the line gets cleared.
- *
- * @param line The current line
+ * @param block The block to convert
+ * @return std::string The converted string
  */
-void ConfigFileParser::processRemainingLine(std::string& line)
+std::string ConfigFileParser::convertBlockToString(Block block) const
 {
-	size_t semicolonIndex = line.find(';');
-	if (semicolonIndex == std::string::npos) {
-		line = "";
-		m_currentLine = line;
-		return;
+	switch (block) {
+	case HttpBlock:
+		return "http";
+	case ServerBlock:
+		return "server";
+	case LocationBlock:
+		return "location";
+	default:
+		return "";
 	}
-
-	line.erase(0, semicolonIndex + 1);
-	line = webutils::trimLeadingWhitespaces(line);
-	webutils::trimTrailingWhiteSpaces(line);
-	m_currentLine = line;
 }
 
 /**
- * @brief Reads the current line of the server config and does several checks
- *
- * How a line gets processed:
- * 1. Gets checked if it contains a semicolon.
- * 2. The directive of the resulting string gets extracted and checked.
- * 3. The value of the directive gets extracted and checked.
+ * @brief Skips the beginning of a block
  *
- * If the line contains more than one directive, this process gets repeated for each directive.
- * Otherwise the line just gets cleared and the processing ends.
+ * @param block The block to skip
  */
-void ConfigFileParser::readServerConfigLine(void)
+void ConfigFileParser::skipBlockBegin(Block block)
 {
-	std::string line = m_currentLine;
-
-	while (!line.empty() && line != "{" && line != "}") {
-		if (isSemicolonMissing(line))
-			throw std::runtime_error("Semicolon missing");
-
-		const std::string directive = getDirective(line);
-		if (directive == "location") {
-			Location location;
-			m_configFile.servers[m_serverIndex].locations.push_back(location);
-			while (m_currentLine != "}") {
-				readAndTrimLine();
-				readLocationConfigLine();
-			}
-			m_locationIndex++;
-			processRemainingLine(line);
-			continue;
-		}
+	while (std::isspace(m_configFileContent[m_configFileIndex]) != 0)
+		m_configFileIndex++;
 
-		const std::string value = getValue(line);
-		if ((value.empty() || value.find_last_not_of(whitespace) == std::string::npos))
-			throw std::runtime_error("'" + directive + "'" + " directive has no value");
-
-		if (!isDirectiveValid(directive, ServerBlock))
-			throw std::runtime_error("Invalid server directive");
+	if (block == HttpBlock)
+		m_configFileIndex += convertBlockToString(HttpBlock).length();
+	else if (block == ServerBlock)
+		m_configFileIndex += convertBlockToString(ServerBlock).length();
+	else if (block == LocationBlock)
+		m_configFileIndex += convertBlockToString(LocationBlock).length();
 
-		readServerDirectiveValue(directive, value);
+	while (std::isspace(m_configFileContent[m_configFileIndex]) != 0)
+		m_configFileIndex++;
 
-		processRemainingLine(line);
+	if (block == LocationBlock) {
```

Comment:
Would call that outside of this function. According to the function name one would expect that only the block denominator is skipped, not also the entire Location block, if it is one. If you had an extra function for this - as mentioned earlier - then you could just call that one.

Reply:
See comment above. The created function gets also used here.